### DISCUSSION TITLE
Graph path finding algorithms

### DISCUSSION THREAD

**Discussion by shivarajmulimani** [cozodb/cozo#48] [Q&A] [answered]
Hi there,
I appreciate the effort and like the project. It’s really good.

I have the following questions on path finding algorithms

context-
1. I have so many users around 10k as user table 
2. I also have one more table which has relations or edges and connects users basically the columns are start , goal, weight, relation type 

question -
1. I want to run path finding algorithm in which I don’t have exact starting user but I know the destination user.
so essentially I want to get all paths which are reachable to a specific destination user.

2. In case if we get paths from all possible starting users to fixed destination user then I want to extract relationship type between each hop for every paths to destination.

**Comment by zh217** [comment, marked as answer]
For 1, you just have to invert your relationships, and then the destination becomes the starting node. Then you can use the usual algorithms.

2 is more complicated. Basically, the algorithm can give you the path, and you need to use your host language to chop the path into path segments, and query the database again to find out the types. Then you can add this information back to the path you extracted. So two trips to the database are required. If you are worried about data inconsistency due to some other processes mutating the database while you run your queries, you can use [multi-statement transaction](https://docs.cozodb.org/en/latest/stored.html#multi-statement-transaction).

**Reply by shivarajmulimani** [reply]
It worked thanks for your inputs.
